config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/PolicyAndFolderLoader.java

```diff
@@ -75,6 +82,24 @@ private void loadPolicies(final File currentDir, final File rootDir, Folder pare
         }
     }
 
+    private Map<Dependency, List<Dependency>> getPolicyDependencies(final File policyRootDir) {
+        final Map<Dependency, List<Dependency>> policyDependencyMap = new HashMap<>();
+        File policyMetadataFile = new File(policyRootDir, "policies" + jsonTools.getFileExtension());
+        final ObjectMapper objectMapper = jsonTools.getObjectMapper();
+        final MapType type = objectMapper.getTypeFactory().constructMapType(HashMap.class, String.class, PolicyMetadata.class);
+        final Map<String, PolicyMetadata> policyMetadataMap = jsonTools.readDocumentFile(policyMetadataFile, type);
+        if (policyMetadataMap != null) {
+            Set<Map.Entry<String, PolicyMetadata>> policyMetadataSet = policyMetadataMap.entrySet();
+            for(Map.Entry<String, PolicyMetadata> metadataEntry : policyMetadataSet){
+                final String policyNameWithPath = metadataEntry.getKey();
+                int index = policyNameWithPath.lastIndexOf("/");
+                final String policyName = index > -1 ? policyNameWithPath.substring(index+1) : policyNameWithPath;
```

Comment:
why do we need to remove folder path from policy name? it will create problem when we compute dependencies for other entities, every time we have to exclude the path.

Reply:
Dependency should have only name not complete path, name is unique identified in dependencies